app/ai/voice/agents/breeze_buddy/services/inbound_policy.py

```
@@ -0,0 +1,211 @@
+        try:
+            is_blacklisted = await is_number_blacklisted(caller_number, merchant_id)
+            if is_blacklisted:
+                logger.info(
+                    f"Inbound call blocked: caller {caller_number} is blacklisted "
+                    f"for merchant {merchant_id}"
+                )
+                return InboundPolicyResult.block(
+                    message="This number is not able to reach us at this time. Goodbye.",
+                    reason="blacklisted",
+                )
+        except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)